tests/utils_tests/test_text.py

```diff
@@ -152,6 +236,10 @@ def test_truncate_html_words(self):
             '<p>The quick <a href="xyz.html"\n id="mylink">brown…</a></p>',
             truncator.words(3, html=True),
         )
+        self.assertEqual(
+            "<p>The…</p>",
+            truncator.words(1, html=True),
+        )
```

Comment:
All of these new tests and assertions should be added up front as a preliminary commit so that we can see what adjustments are made in the main commit when migrating to the new parser.

Reply:
I _think_ I've already done this as these tests are in this commit here 🤔  https://github.com/django/django/pull/16421/commits/fdc15362d6706ff6cef7927b6b6617a9cce8a1b8 

I'll also work on adding Adam's `"foo</p>"` test to this up front commit as well.

Reply:
Sorry, yes, noticed this after the review 👍🏻